prober.py

```diff
@@ -71,17 +72,22 @@ def does_build_succeed(filename, linker_options=""):
 def compile_and_run(filename, linker_options=""):
     # Utility function that returns the stdout output from running the
     # compiled source file; None if the compile fails.
-    cmd = "cc -Wall -o ./prober/foo %s ./prober/%s" % (linker_options, filename)
+    cc = os.getenv("CC", "cc")
+    cmd = "%s -Wall -o ./prober/foo %s ./prober/%s" % (cc, linker_options, filename)
 
     p = subprocess.Popen(cmd, shell=True, stdout=STDOUT, stderr=STDERR)
 
     if p.wait():
         # uh-oh, compile failed
         return None
-    else:
+    
+    try:
         s = subprocess.Popen(["./prober/foo"],
                              stdout=subprocess.PIPE).communicate()[0]
         return s.strip().decode()
+    except:
```

Comment:
```suggestion
    except Exception:
```

This is better form, according to PEP 8 (https://peps.python.org/pep-0008/)

Reply:
Yeah, I really don't think that this is a controversial suggestion... do you want me to squash the commits and rebase?